dask_planner/src/sql/logical/table_scan.rs

```diff
@@ -18,6 +19,129 @@ pub struct PyTableScan {
+    }
+
+    /// Transform the singular Expr instance into its DNF form serialized in a Vec instance. Possibly recursively expanding
+    /// it as well if needed.
+    pub fn _expand_dnf_filter(
+        filter: &Expr,
+        py: Python,
+    ) -> Result<Vec<(String, String, Vec<PyObject>)>, DaskPlannerError> {
+        let mut filter_tuple: Vec<(String, String, Vec<PyObject>)> = Vec::new();
+
+        match filter {
+            Expr::InList {
+                expr,
+                list,
+                negated,
+            } => {
+                // Only handle simple Expr(s) for InList operations for now
+                if PyTableScan::_valid_expr_type(list) {
+                    // While ANSI SQL would not allow for anything other than a Column or Literal
+                    // value in this "identifying" `expr` we explicitly check that here just to be sure.
+                    // IF it is something else it is returned to Dask to handle
+                    let ident = match *expr.clone() {
+                        Expr::Column(col) => Ok(col.name),
+                        Expr::Literal(val) => Ok(format!("{}", val)),
+                        _ => Err(DaskPlannerError::InvalidIOFilter(format!(
+                            "Invalid InList Expr type `{}`. using in Dask instead",
+                            filter
+                        ))),
+                    };
+
+                    let op = if *negated { "not in" } else { "in" };
+                    let il: Vec<PyObject> = list
+                        .iter()
+                        .map(|f| match f {
+                            Expr::Column(col) => col.name.clone().into_py(py),
+                            Expr::Literal(val) => match val {
+                                ScalarValue::Boolean(val) => val.unwrap().into_py(py),
+                                ScalarValue::Float32(val) => val.unwrap().into_py(py),
+                                ScalarValue::Float64(val) => val.unwrap().into_py(py),
+                                ScalarValue::Int8(val) => val.unwrap().into_py(py),
+                                ScalarValue::Int16(val) => val.unwrap().into_py(py),
+                                ScalarValue::Int32(val) => val.unwrap().into_py(py),
+                                ScalarValue::Int64(val) => val.unwrap().into_py(py),
+                                ScalarValue::UInt8(val) => val.unwrap().into_py(py),
+                                ScalarValue::UInt16(val) => val.unwrap().into_py(py),
+                                ScalarValue::UInt32(val) => val.unwrap().into_py(py),
+                                ScalarValue::UInt64(val) => val.unwrap().into_py(py),
+                                ScalarValue::Utf8(val) => val.clone().unwrap().into_py(py),
+                                ScalarValue::LargeUtf8(val) => val.clone().unwrap().into_py(py),
+                                _ => "".into_py(py),
```

Comment:
For an unsupported scalar type we should raise an error and have it go via the unfiltered_exprs path